### DISCUSSION TITLE
How to sync a single directory in docker?

### DISCUSSION THREAD

**Discussion by ukreddy-erwin** [abraunegg/onedrive#1700] [Q&A] [answered]
I am trying to use the default docker image.

https://github.com/abraunegg/onedrive/blob/master/docs/Docker.md

But this is syncing all the folders but I require only a particular folder and sub folders of it to sync.

Earlier I used the direct binary where I use the below command to sync a single directory from business shared folders.
`
onedrive --synchronize --sync-shared-folders --single-directory 'Jenkins' --verbose`

And the configurations added:
```
cat ~/.config/onedrive/business_shared_folders

Jenkins

cat ~/.config/onedrive/business_shared_folders

Jenkins/LatestBuilds
```

**Comment by abraunegg** [comment, marked as answer]
@ukreddy-erwin 
You cannot use `--single-directory` with Docker - it is incompatible.

To sync a single directory | specific folder structure you must use 'sync_list' in a Docker environment.

**Reply by ukreddy-erwin** [reply]
In the docker volume for configurations, I have added below file.

business_shared_folders

with content:
Jenkins/LatestBuilds

Is this you are referring to ?

I didn't find any reference to sync_list in your docker documentation for shared folders but I gave like this.

```
ubuntu@ip-172-31-24-95:/var/lib/docker/volumes/onedrive_conf/_data$ ls
business_shared_folders  items.sqlite3  items.sqlite3-shm  items.sqlite3-wal  refresh_token  sync_list
ubuntu@ip-172-31-24-95:/var/lib/docker/volumes/onedrive_conf/_data$ cat sync_list
Jenkins/LatestBuilds

sync_business_shared_folders = "true"
```


But it seems it treating Jenkins as normal folder instead of shared folder and not syncing anything.

**Reply by abraunegg** [reply]
No - you need to add a new file called 'sync_list'.

Read the usage document for details

**Reply by ukreddy-erwin** [reply]
Yes I created a separate file name sync_list in the config folder and added content as below but this time it not syncing anything, I guess as there is no local folder Jenkins, the Jenkins is a shared folder.

```
cat sync_list
Jenkins/LatestBuilds
sync_business_shared_folders = "true"
```

**Reply by abraunegg** [reply]
@ukreddy-erwin 
This is partially correct .. please re-read the usage document and correct your configuration

**Reply by ukreddy-erwin** [reply]
I didn't find any commadn related to shared-folders in sync_list

**Reply by abraunegg** [reply]
Please read the usage.md document as per https://github.com/abraunegg/onedrive/blob/master/docs/USAGE.md#selective-sync-via-sync_list-file

**Reply by ukreddy-erwin** [reply]
my sync_list file, still not getting the files:

Jenkins/LatestBuilds
sync_business_shared_folders = "true"

**Reply by abraunegg** [reply]
@ukreddy-erwin 
Please read the documentation ... you are not reading it correctly

Once you have configured the application:

* Use `onedrive --display-config` to list your configuration
* Use `onedrive --dry-run` to test your configuration
